src/components/ListItem.jsx

```diff
@@ -1,5 +1,58 @@
 import './ListItem.css';
+import { useState, useEffect } from 'react';
+import { updateItem } from '../api/firebase';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({
+	name,
+	itemId,
+	dateLastPurchased,
+	totalPurchases,
+	listId,
+}) {
+	const [isChecked, setIsChecked] = useState(false);
+
+	const increasePurchases = totalPurchases + 1;
+	const decreasePurchases = totalPurchases - 1;
+
+	useEffect(() => {
+		const currentTime = Date.now();
+
+		if (dateLastPurchased === null) {
+			return;
+		}
+		const timeSinceLastPurchase = currentTime - dateLastPurchased.toMillis();
+
+		if (timeSinceLastPurchase < 24 * 60 * 60 * 1000) {
+			setIsChecked(true);
+		} else {
+			setIsChecked(false);
+		}
+	}, [dateLastPurchased]);
+
+	const handleCheck = async () => {
```

Comment:
I refactored the `handleCheck` function like so for better readability: 

```
const handleCheck = async () => {
		setIsChecked(!isChecked);
		
		await updateItem(listId, itemId, {
			dateLastPurchased: !isChecked ? new Date() : null,
			totalPurchases: !isChecked ? increasePurchases : decreasePurchases,
			});
	};
```

Reply:
Thanks for the input, we will play around with this function.

Yes, the goal was to allow the user to uncheck a box and have our database update correctly based on that action. As a user, you could click a checkbox by mistake or change your mind after clicking it - so we needed a way to account for this by "resetting" the state & database.

Reply:
We refactored this function, combining the suggestions and our own code preferences. Let us know if the changes are more understandable.

Reply:
Thank you Yufa. That's why we decided to remove that functionality. We were trying to account for accidental checkmarks placed by the user because we thought it was part of the acceptance criteria. Upon closer examination, and talking it over with the team, we realized it wasn't necessary.

Reply:
Thank you Yufa. That's why we decided to remove that functionality. We were trying to account for accidental checkmarks placed by the user because we thought it was part of the acceptance criteria. Upon closer examination, and talking it over with the team, we realized it wasn't necessary.